zebedee-cms/src/test/java/com/github/onsdigital/zebedee/permissions/service/PermissionsServiceImplTest.java

```diff
@@ -407,59 +279,201 @@ public void addAdministrator_ShouldThrowErrorIfUserNotAdmin() throws Exception {
         try {
             permissions.addAdministrator(email2, session);
         } catch (UnauthorizedException e) {
-            verify(permissionsStore, times(2)).getAccessMapping();
-            verify(accessMapping, times(4)).getAdministrators();
-            verifyNoMoreInteractions(permissionsStore, accessMapping, teamsService);
+            verify(permissionsStore, times(1)).getAccessMapping();
+            verify(accessMapping, times(2)).getAdministrators();
+            verifyNoMoreInteractions(permissionsStore, accessMapping);
             throw e;
         }
     }
 
 
     @Test
     public void addAdministrator_Success() throws Exception {
+        Set<String> admins = new HashSet<>();
         admins.add(EMAIL);
         String email2 = "test2@ons.gov.uk";
+        List<String> sessionGroups = new ArrayList<>();
+        sessionGroups.add(ADMIN_GROUP);
 
+        when(session.getGroups())
+                .thenReturn(sessionGroups);
         when(permissionsStore.getAccessMapping())
                 .thenReturn(accessMapping);
         when(accessMapping.getAdministrators())
                 .thenReturn(admins);
 
         permissions.addAdministrator(email2, session);
 
-        assertEquals(2, admins.size());
         assertTrue(admins.contains(email2));
-        verify(permissionsStore, times(3)).getAccessMapping();
-        verify(accessMapping, times(6)).getAdministrators();
+        verify(permissionsStore, times(2)).getAccessMapping();
+        verify(accessMapping, times(4)).getAdministrators();
         verify(permissionsStore, times(1)).saveAccessMapping(accessMapping);
     }
 
+    @Test(expected = UnauthorizedException.class)
+    public void addEditor_ShouldThrowErrorSessionNull() throws Exception {
+        Set<String> admins = new HashSet<>();
+        admins.add(EMAIL);
+
+        when(permissionsStore.getAccessMapping())
+                .thenReturn(accessMapping);
+        when(accessMapping.getAdministrators())
+                .thenReturn(admins);
+
+        try {
+            permissions.addEditor(EMAIL, null);
+        } catch (UnauthorizedException e) {
+            verify(permissionsStore, times(1)).getAccessMapping();
+            verify(accessMapping, times(2)).getAdministrators();
+            verifyNoMoreInteractions(permissionsStore, accessMapping);
+            throw e;
+        }
```

Comment:
No need to change this but for reference a nicer way to test exception cases is to use `assertThrows()`. It takes the expected exception class and the method to run and will fail the test if an exception of that type is not thrown. Its a bit nicer than try catch throw again.

```
UnauthorizedException ex = assertThrows(UnauthorizedException.class, () -> permissions.addEditor(EMAIL, null));

verify(permissionsStore, times(1)).getAccessMapping();
verify(accessMapping, times(2)).getAdministrators();
verifyNoMoreInteractions(permissionsStore, accessMapping);
```